### PR TITLE
Feature: Added one Github workflow to auto update hugo theme

### LINKED ISSUE #2436: Update the website theme and document the process
### Is your feature request related to a problem?

In the code we have been using https://github.com/McShelby/hugo-theme-relearn for the website, it would be nice to have an easy process to keep the theme updated

### Describe the solution you'd like

Maybe github action that creates a PR, or an inv command, to be defined.

### Additional context

_No response_

### I've read and agree with the following

- [X] I've checked all open and closed issues and my request is not there.
- [X] I've checked all open and closed pull requests and my request is not there.

### PR DESCRIPTION
**Pull request by ayushrakesh**
**Feature - Added one github action to automatically update hugo-theme weekly**

**Fixes #2436 **:

> Uncomment only one, leave it on its own line:
>
> /kind bug
> /kind cleanup 
   /kind feature
> /kind design
> /kind flake
> /kind failing
> /kind documentation
> /kind regression

**What is does - Automates the upgradation of hugo-theme by fetching the latest theme. Added one github workflow.**

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: .github/workflows/hugo-theme-update.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+
+      - name: Update Hugo Theme
+        run: |
+          cd themes
+          git clone https://github.com/McShelby/hugo-theme-relearn.git
```
[the remark this answers is not in the document]
Comment (ayushrakesh) [reply .github/workflows/hugo-theme-update.yaml:19]: @karampok thanks for suggestion, you mean to watch for latest release and then make PR for it?

[the remark this answers is not in the document]
Comment (karampok) [reply .github/workflows/hugo-theme-update.yaml:19]: Yes, that is what I am thinking. On every release of the theme only to create a PR that in the title/commit title is the release vesrion, and in the description/commit description the release notes.

[the remark this answers is not in the document]
Comment (ayushrakesh) [reply .github/workflows/hugo-theme-update.yaml]: @karampok I have revised the PR, please chekout it out, btw thanks for help.

[the remark this answers is not in the document]
Comment (karampok) [reply .github/workflows/hugo-theme-update.yaml]: in theory you do not need to git clone, you can leverage that the release has the code you need, no need for git commands.

Maybe (never personally checked) we could leverage a standard gh action https://github.com/marketplace/actions/release-downloader?

Also ideally we need to stamp the release in the repo, and only create if there is a new one.

[the remark this answers is not in the document]
Comment (ayushrakesh) [reply .github/workflows/hugo-theme-update.yaml]: @karampok I need help, should we have to use to standard Github action for creating PR? Not experienced with it. Any suggestions?

[the remark this answers is not in the document]
Comment (fedepaol) [reply .github/workflows/hugo-theme-update.yaml]: you can take inspiration from this job https://github.com/metallb/metallb-operator/blob/main/.github/workflows/bump_metallb.yml

[the remark this answers is not in the document]
Comment (ayushrakesh) [reply .github/workflows/hugo-theme-update.yaml]: @fedepaol @karampok PR revised.

**Comment by ayushrakesh** [comment]
@fedepaol @karampok Please review my PR.

**Comment by ayushrakesh** [comment]
@fedepaol @karampok Please review my PR. Any suggestions would be grateful.

File: .github/workflows/hugo-theme-update.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        if: steps.check-release.outputs.new_release == 'true'
+        run: |
+          curl -L https://github.com/McShelby/hugo-theme-relearn/archive/refs/tags/${{ steps.latest-release.outputs.release_tag }}.tar.gz | tar xz -C repo/themes/hugo-theme-relearn --strip-components=1
+
+      - name: Commit and push changes
+        if: steps.check-release.outputs.new_release == 'true'
+        run: |
+          cd repo
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
```
Comment (fedepaol) [inline .github/workflows/hugo-theme-update.yaml:58]: where does this come from?

**Comment by fedepaol** [comment]
@ayushrakesh apologies for the delay in reviewing.
The changes look good, I have one question about it. Additionally, did you test this in some way?

**Pull request closed without merging**